### ISSUE TITLE
MA0076 does not respect "null" invariant format

### ISSUE THREAD

**Issue by drieseng** [meziantou/Meziantou.Analyzer#897]
### Version of the Meziantou.Analyzer NuGet package

2.0.231

### Rule Identifier

MA0076

### Target Framework

.NET 9.0

### C# Language version

4.14.0-3.25412.6 (c167adef)'. Language version: 13.0

### Description

When the **null** is configured as being invariant for a given (custom) type (using **CultureInsensitiveTypeAttribute**), then the MA0076 diagnostic is still reported.

### Reproduction Steps

```c#
using System;
using Meziantou.Analyzer.Annotations;

internal sealed class Program
{
    private static void Main(string[] args)
    {
        Foo foo = new Foo();

        Console.WriteLine(foo.ToString(format: null)); // This is OK
        Console.WriteLine($"{foo:null}"); // MA0076 should not be reported as it's "annotated" as being invariant
    }
}

[CultureInsensitiveType(format: null)]
public sealed class Foo : IFormattable
{
    public string ToString(string format)
    {
        return string.Empty;
    }

    public string ToString(IFormatProvider formatProvider)
    {
        return string.Empty;
    }

    public string ToString(string format, IFormatProvider formatProvider)
    {
        return string.Empty;
    }

    public override string ToString()
    {
        return string.Empty;
    }
}
```

**Comment by meziantou** [comment]
> Console.WriteLine($"{foo:null}"); // MA0076 should not be reported as it's "annotated" as being invariant

This is equivalent to `Console.WriteLine(foo.ToString("null"))`. The `"null"` format is not marked as invariant, so it's reported.

**Comment by drieseng** [comment]
I'm gonna crawl under a rock :p
